### PR TITLE
[WIP][ADD] base_geoengine_jsonify: support of geo_point json exports

### PR DESCRIPTION
**Pull request by len-foss**
Dependent on https://github.com/OCA/server-tools/pull/1894 refactoring to actually work.

### CODE REVIEW

File: base_geoengine_jsonify/models/models.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+from odoo import models, tools
+
+
+class Base(models.AbstractModel):
+    _inherit = "base"
+
+    def _jsonify_value(self, field, value):
+        """Adds support for geo_point fields."""
+        res = super(Base, self)._jsonify_value(field, value)
+        if res and field.type == "geo_point":
```
Comment (lmignon) [inline base_geoengine_jsonify/models/models.py]: @len-foss The same logic could be applied to all the geo fields non? 

```python

from  odoo.addons.base_geoengine.fields import GeoField

....

if res and isinstance(field, GeoField):
    res = tools.safe_eval(geojson.dumps(res))
...

```

**Review by lmignon — APPROVED** [review]
Great!
LGTM once OCA/server-tools#1894  is merged

File: base_geoengine_jsonify/models/models.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+class Base(models.AbstractModel):
+    _inherit = "base"
+
+    def _jsonify_value(self, field, value):
+        """Adds support for geo_point fields."""
+        res = super(Base, self)._jsonify_value(field, value)
+        if res and isinstance(field, GeoField):
+            # we eval it as dumps gives a str, we need a point for json.dumps
+            res = tools.safe_eval(geojson.dumps(res))
```
Comment (sbidoul) [inline base_geoengine_jsonify/models/models.py]: If `geojson.dumps` produces a json str, is it not better to parse it with `json.loads` ?

**Comment by yvaucher** [comment]
OCA/server-tools#1894 was merged on Jan 21

**Review by yvaucher — APPROVED** [review]
LGTM

Looks interesting

File: base_geoengine_jsonify/models/models.py (hunk 2 above)
Comment (yvaucher) [reply base_geoengine_jsonify/models/models.py]: This seems to have been resolved with:
 
 ``` python
           # we eval it as dumps gives a str, we need a point for json.dumps
            res = geojson.loads(geojson.dumps(res))
```

Seems weird, waiting on travis results (don't know what to expect from the content of `res` in this context)

**Comment by yvaucher** [comment]
Just relaunched the CI

File: base_geoengine_jsonify/models/models.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+from odoo import models
+from  odoo.addons.base_geoengine.fields import GeoField
+
+
+class Base(models.AbstractModel):
+    _inherit = "base"
+
+    def _jsonify_value(self, field, value):
+        """Adds support for geo_point fields."""
```
Comment (yvaucher) [inline base_geoengine_jsonify/models/models.py:14]: Let's be a bit more generic, geojson.dumps should support multiline string and polygones too.

```suggestion
        """Adds support for geometry fields."""
```

File: base_geoengine_jsonify/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Automatically export geopoint fields as geojson in json exports.
```
Comment (yvaucher) [inline base_geoengine_jsonify/readme/DESCRIPTION.rst:1]:
```suggestion
Automatically export geometry fields as geojson in json exports.
```

File: base_geoengine_jsonify/models/models.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        res = super(Base, self)._jsonify_value(field, value)
+        if res and isinstance(field, GeoField):
+            # we eval it as dumps gives a str, we need a point for json.dumps
```
Comment (yvaucher) [inline base_geoengine_jsonify/models/models.py:17]:
```suggestion
            # we eval it as dumps gives a str, we need a geometry for json.dumps
```

File: base_geoengine_jsonify/readme/USAGE.rst [hunk 6]
```diff
@@ -0,0 +1,5 @@
+Nothing to do!
+If you add a geo_point to a json export, the point will be converted automatically
```
Comment (yvaucher) [inline base_geoengine_jsonify/readme/USAGE.rst:2]:
```suggestion
If you add a geometry to a json export, the point will be converted automatically
```

File: base_geoengine_jsonify/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2020 ACSONE SA/NV.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Support of GeoPoint Json exports for Odoo",
```
Comment (yvaucher) [inline base_geoengine_jsonify/__manifest__.py:4]:
```suggestion
    "name": "Export geometry as GeoJSON",
```

**Review by yvaucher — APPROVED** [review]
Nice, few suggestion to make the module a bit more generic in the terminology as GeoPoint is not the only geometry that can be handled. I assume it would work too, to create a GeoJSON for a Polygon or a LineString.

**Pull request closed without merging**